support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -236,9 +245,14 @@ export function Checkout({ geoId, appConfig, abParticipations }: Props) {
 	const forcedCountry = urlSearchParams.get('country') ?? undefined;
 
 	/**
-	 * The returnLink for back button and thank you page
+	 * returnAddress re-added for future use on thank-you page
 	 */
-	const returnLink = urlSearchParams.get('returnAddress') ?? undefined;
+	const sessionStorageReturnAddress = storage.session.get('returnAddress');
+	const parsedOrder = safeParse(
+		ReturnAddressSchema,
+		sessionStorageReturnAddress,
+	);
+	const returnLink = parsedOrder.success ? parsedOrder.output.link : undefined;
```

Comment:
Do we need to parse the `returnAddress` on the TY page URL now we're saving it in session storage, the TY page should be able to read it from session storage now?

Reply:
See comment below, its now been removed completely from `checkout.tsx`, updated